interview-questions.md

```diff
@@ -8,50 +8,92 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-Researched answer:
 
-5. What is the difference between a class and an object?
 
-Your answer:
+5. What is the difference between a class and an object?
 
-Researched answer:
+Your answer: Objects are collections of key:value pairs which are stored inside curly braces. Class is a function which is used to create a template, or blueprint, for making objects. Classes can also includes methods, which are functions contained within and belonging to that class. For example, if I needed to create many objects such as products in an online shop, I could first create a class called Product which would use the method constructor to take parameters of the defining properties all of the products would have, such as brand, type, price, category, etc. Then, to create the objects, I could use the keyword class to quickly and easily set up many different objects.
+
+class Product = {
+    constructor(brand, type, price, category){
+        this.brand = brand
+        this.type = type
+        this.price = price
+        this.category = category
+    }
+    describe() = {return `In the ${this.category} section, we have a ${this.brand} {this.type} for the low price of $${this.price}!`
+}
+let keurig = new Product("Keurig", "coffee maker", "65", "Applicances")
+let hottub = new Product("Whirlpool", "jacuzzi", "1500", "Outdoor")
+
+Researched answer: When learning more about classes, I learned about the idea of hoisting. This refers to what order JS reads the code and performs it. While some functions can be invoked before they are actually defined, classes do not fall into this category. First, a class must be declared, and then, below, they can be used to construct objects.
+I also learned about class expressions which are a way of defining classes. For example, if there are going to be several similar but distinct classes, it may be helpful to group them within a class expression. The syntax would be:
+Instrument = class Guitar {
+    constructor(brand, type){
+        this.brand = brand
+        this.type = type
+    }
+
+}
 
 6. STRETCH: What is hoisting in JavaScript?
 
-Your answer:
+Your answer: Wow I did not read ahead in the questions or I might have written something else in my previous question!
```

Comment:
Crushed this section!!  All of your answers and researched answers are thorough and very well worded.  Great job!  🎉